packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -66,6 +67,7 @@ export class OdspDocumentService implements IDocumentService {
         socketIoClientFactory: () => Promise<SocketIOClientStatic>,
         cache: IOdspCache,
         hostPolicy: HostStoragePolicy,
+        fetcher: Fetcher | undefined,
```

Comment:
if possible, it would be great to avoid optional arguments. I.e. either always create them at higher level (factory), or lowest (driver instance). I think later makes more sense - it can be always created in constructor, not sure why factory needs to be involved.

Reply:
The file is create in factory not in driver service. So in that case, it needs to created in factory and then passed to createDocumentService. So I think in both cases, I will create it in factory.

Reply:
Create new call happens before creating the document service.

Reply:
I see - this makes sense in general. But right here, we get only from createDocumentService() for exising file, right? So it's always undefined here.

Reply:
Removed passing the epoch in constructor. Also removed undefined from the type.